scripts/generate_pyapi_stubs/generate_pyapi_stubs.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# Copyright (C) 2018-2025 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+invalid_expressions=(
+    # The classes bindings' will be soon added to pyopenvino
+    "ov::op::v1::Add"
+    "ov::op::v1::Divide"
+    "ov::op::v1::Multiply"
+    "ov::op::v1::Subtract"
+    "ov::op::v1::Divide"
+
+    # Circular dependency Input/Output/Node/Tensor
+    "ov::Node"
+    "ov::Input<ov::Node>"
+    "ov::descriptor::Tensor"
+    "<Type: 'undefined'>"
+    "ov::Output<ov::Node const>"
+
+    # New bindings required
+    "ov::float16"
+    "ov::EncryptionCallbacks"
+    "ov::streams::Num"
+
+    # Other issues
+    "<Dimension:"
+)
+
+invalid_identifiers=(
+    # Ticket:
+    "<locals>"
+)
+
+unresolved_names=(
+    # Circular dependencies
+    "InferRequestWrapper"
+    "RemoteTensorWrapper"
+    "capsule"
+    "VASurfaceTensorWrapper"
+
+    # Other issues
+    "_abc._abc_data"
+)
+
+create_regex_pattern() {
+    local errors=("$@")
+    local regex_pattern=""
+
+    for error in "${errors[@]}"; do
+        escaped_error=$(printf '%s\n' "$error" | sed -e 's/[]\/$*.^|[]/\\&/g')
+        regex_pattern+=".*$escaped_error.*|"
+    done
+    regex_pattern="${regex_pattern%|}"
+    echo "$regex_pattern"
+}
+
+invalid_expressions_regex=$(create_regex_pattern "${invalid_expressions[@]}")
+invalid_identifiers_regex=$(create_regex_pattern "${invalid_identifiers[@]}")
+unresolved_names_regex=$(create_regex_pattern "${unresolved_names[@]}")
+output_dir="$(dirname "$0")/../../src/bindings/python/src"
+
+python -m pybind11_stubgen \
+            --output-dir "$output_dir" \
+            --root-suffix "" \
+            --ignore-invalid-expressions "$invalid_expressions_regex" \
+            --ignore-invalid-identifiers "$invalid_identifiers_regex" \
+            --ignore-unresolved-names "$unresolved_names_regex" \
+            --numpy-array-use-type-var \
+            --exit-code \
+            openvino
```

Comment:
BTW, how do you manage to generate multiple .pyi fiels?
In GenAI we have the only one, but maybe it's good to split into several ones.

Reply:
It's not a design choice, but rather the way `pybind11-stubgen` works. In case of GenAI it apparently creates a single `.pyi` file (presumably due to it being a single module), in case of OpenVINO it creates a directory structure full of files, resembling our module structure.